proj_eclipse/src/main/java/smartmag/models/ProductModel.java

```diff
@@ -68,8 +83,28 @@ public void createProdotto()
 		copyProdottoIntoRecord(prodotto, r);
 		r.store(); // INSERT
 		this.record = r;
+		notifyChangeListeners(null);
+	}
 
-		// TODO: event
+	public Prodotto getProdotto() {
+		return prodotto.clone();
+	}
+
+	protected void setProdotto(Prodotto p) {
+
+		if (p != null && p.isValid())
+			this.prodotto = p;
```

Comment:
Clone per evitare modifiche inattese
```suggestion
			this.prodotto = p.clone();
```